models/seir/model_mod.f90

```diff
@@ -0,0 +1,455 @@
+! DART software - Copyright UCAR. This open source software is provided
+! by UCAR, "as is", without charge, subject to all terms of use at
+! http://www.image.ucar.edu/DAReS/DART/DART_download
+!
+
+module model_mod
+
+use types_mod,             only : r8, i8, i4, MISSING_R8
+
+use time_manager_mod,      only : time_type, set_time
+
+use location_mod,          only : location_type, set_location, get_location,  &
+                                  get_close_obs, get_close_state,             &
+                                  convert_vertical_obs, convert_vertical_state
+
+use utilities_mod,         only : register_module, do_nml_file, do_nml_term,    &
+                                  nmlfileunit, find_namelist_in_file,           &
+                                  check_namelist_read
+
+use location_io_mod,      only :  nc_write_location_atts, nc_write_location
+
+use netcdf_utilities_mod, only : nc_add_global_attribute, nc_synchronize_file, &
+                                 nc_add_global_creation_time, nc_begin_define_mode, &
+                                 nc_end_define_mode
+
+use         obs_kind_mod,  only : QTY_STATE_VARIABLE
+
+use  mpi_utilities_mod,    only : my_task_id
+
+use random_seq_mod,        only : random_seq_type, init_random_seq, random_gaussian
+
+use ensemble_manager_mod,  only : ensemble_type, get_my_num_vars, get_my_vars
+
+use distributed_state_mod, only : get_state
+
+use state_structure_mod,   only : add_domain
+
+use default_model_mod,     only : end_model, nc_write_model_vars
+
+use dart_time_io_mod,      only : read_model_time, write_model_time
+
+implicit none
+private
+
+public :: get_model_size,       &
+          get_state_meta_data,  &
+          model_interpolate,    &
+          shortest_time_between_assimilations, &
+          static_init_model,    &
+          init_conditions,      &
+          init_time,            &
+          adv_1step,            &
+          nc_write_model_atts
+
+public :: pert_model_copies,      &
+          nc_write_model_vars,    &
+          get_close_obs,          &
+          get_close_state,        &
+          end_model,              &
+          convert_vertical_obs,   &
+          convert_vertical_state, &
+          read_model_time, &
+          write_model_time
+
+! version controlled file description for error handling, do not edit
+character(len=256), parameter :: source   = "new_model.f90"
+
+type(location_type), allocatable :: state_loc(:)  ! state locations, compute once and store for speed
```

Comment:
Only need single value, because location is 0.5 for all state. See get_state_meta_data comment. 
```suggestion
type(location_type) :: state_loc ! state location, compute once and store for speed
```

Reply:
Excellent catch!